compiler/rustc_middle/src/mir/coverage.rs

```diff
@@ -41,8 +41,14 @@ rustc_index::newtype_index! {
 }
 
 impl CounterValueReference {
-    // Counters start at 1 to reserve 0 for ExpressionOperandId::ZERO.
+    /// Counters start at 1 to reserve 0 for ExpressionOperandId::ZERO.
     pub const START: Self = Self::from_u32(1);
+
+    /// Returns explicitly-requested zero-based version of the counter id, used
+    /// during codegen. LLVM expects zero-based indexes.
+    pub fn zero_based_index(&self) -> u32 {
+        self.as_u32() - 1
```

Comment:
If this function isn't too performance sensitive, its probably worthwhile to `.checked_sub(1).unwrap_or(|| bug!(...))` to safeguard against `ZERO.zero_based_index()` being invoked.

Reply:
Now that I'm calling this from `counter_value_reference()` (per your first comment), I moved the `debug_assert` here, so it will still be checked for bugs. Thanks!

Reply:
done